crypto/asn1/asn1_gen.c

```diff
@@ -563,11 +562,10 @@ static int asn1_str2tag(const char *tagstr, int len)
     if (len == -1)
         len = strlen(tagstr);
 
-    tntmp = tnst;
-    for (i = 0; i < OSSL_NELEM(tnst); i++, tntmp++) {
-        if ((len == tntmp->len)
-            && (OPENSSL_strncasecmp(tntmp->strnam, tagstr, len) == 0))
-            return tntmp->tag;
+    const struct tag_name_st *tp = tnst;
+    for (i = OSSL_NELEM(tnst); --i >= 0; tp++) {
```

Comment:
Not a big deal, but I find it much easier to understand this loop the way this was before. Why go backwards?

Reply:
old habits; counting down sets the CPU `Z` flag for free :). fixup pushed that undid it.